src/Sarif/ExtensionMethods.cs

```diff
@@ -182,56 +182,60 @@ public static string GetMessageText(this Result result, IRule rule, bool concise
             if (string.IsNullOrEmpty(text))
             {
                 text = string.Empty;    // Ensure that it's not null.
-
+                
                 string messageId = result.Message?.MessageId;
+                string[] arguments = null;
 
-                if (rule != null && !string.IsNullOrWhiteSpace(messageId))
+                if (result.Message?.Arguments != null)
+                {
+                    arguments = new string[result.Message.Arguments.Count];
+                    result.Message.Arguments.CopyTo(arguments, 0);
+                }
+                else
                 {
-                    string messageString;
+                    arguments = new string[0];
+                }
```

Comment:
You may as well defer constructing the `arguments` array until you know whether you have a message format string. I'd move this into the `if` block starting on Line 199.

Reply:
Left over from the refactor...

---
In reply to: [212701501](https://github.com/Microsoft/sarif-sdk/pull/1048#discussion_r212701501) [](ancestors = 212701501)

Reply:
Actually you also don't need to fetch the messageId if you don't even have a rule. Most efficient (and, IMO, clearest) is:
```
Fetch message text
IF message text is null or empty
    IF there is a rule
        Fetch message id
        IF there is a message id AND there is a rule AND the rule has a message with that id
      
    ELSE
        Set text to empty string.
```

---
In reply to: [212702127](https://github.com/Microsoft/sarif-sdk/pull/1048#discussion_r212702127) [](ancestors = 212702127,212701501)

Reply:
Looks good

---
In reply to: [212703463](https://github.com/Microsoft/sarif-sdk/pull/1048#discussion_r212703463) [](ancestors = 212703463,212702127,212701501)

Reply:
I left the text = "" line where it is so we never return null

---
In reply to: [212704250](https://github.com/Microsoft/sarif-sdk/pull/1048#discussion_r212704250) [](ancestors = 212704250,212703463,212702127,212701501)

Reply:
Sorry, what is the "this" that returns null?

---
In reply to: [212704250](https://github.com/Microsoft/sarif-sdk/pull/1048#discussion_r212704250) [](ancestors = 212704250,212703463,212702127,212701501)

Reply:
Oh, did I miss a case. Ok.

---
In reply to: [212705797](https://github.com/Microsoft/sarif-sdk/pull/1048#discussion_r212705797) [](ancestors = 212705797,212704250,212703463,212702127,212701501)

Reply:
Sorry, if this fails: IF there is a message id AND the rule has a message with that id
text will be null

---
In reply to: [212705834](https://github.com/Microsoft/sarif-sdk/pull/1048#discussion_r212705834) [](ancestors = 212705834,212704250,212703463,212702127,212701501)